userspace/libsinsp/metrics_collector.cpp

```diff
@@ -91,6 +90,41 @@ std::string metric_value_to_text(const metrics_v2& metric)
 	return value_text;
 }
 
+std::string prometheus_sanitize_metric_name(const std::string& name)
+{
+	// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+	static const RE2 invalid_chars("[^a-zA-Z0-9_]");
+	std::string sanitized_name = name;
+	RE2::GlobalReplace(&sanitized_name, invalid_chars, "_");
+	RE2::GlobalReplace(&sanitized_name, "_+", "_");
+	if (!sanitized_name.empty() && !(std::isalnum(sanitized_name.front()) || sanitized_name.front() == '_'))
+	{
+		sanitized_name = "_" + sanitized_name;
+	}
+	if (!sanitized_name.empty() && sanitized_name.back() == '_')
+	{
+		sanitized_name.pop_back();
+	}
```

Comment:
Misread the regex and thought for the metric name also numbers are allowed in front. Plus yes the `!sanitized_name.empty()` was an error here. Re-pushed to 100% strictly stay with the prometheus mandates. 

Right now we would replace and empty metrics name with `_`. We chatted a bit over slack and are unsure how to best handle it. Logs could be too noisy and since metrics are not that critical, we certainly don't want to error out or anything. Maybe simply not adding the metric is the right solution.